share/revng/test/configuration/revng-qa/for-runtime-abi-tests.yml

```diff
@@ -68,28 +68,61 @@ tags:
     variables:
       ABI_NAME: Microsoft_x86_cdecl
       GCC_CFLAGS:
-        - -DABIDEF=__attribute__((ms_abi,cdecl))
+        # safeguard in case somebody tries using it with gcc
+        - -DABIDEF=UNSUPPORTED
       MSVC_CFLAGS:
         - -DABIDEF=__cdecl
 
+  - name: i386-cdecl-microsoft-abi-gcc-edition
```

Comment:
Why do we test GCC at all given that we have MSVC?

Given how badly GCC implements MS ABIs, it feels like a waste of time.
Let's try to save on test runtimes, unless there's a real benefit.

Reply:
> Why do we test GCC at all given that we have MSVC?

64-bit one is unusably bad. So I wouldn't even think about trying to put that one through, but 32-bit (even if incompatible) are kind of usable. And it doesn't require any extra effort to support those (they don't do anything weird or unusual - kind of the same as `regpart` ABIs but with msvc-like registers used).

Either way, I'm fine with either keeping them, removing or, maybe, just commenting them out. That way we can still preserve them if we ever run into related problems (I'd imagine there is at least some code out there in the wild compiled by GCC in "msvc-abi" mode, hell, MinGW exists), but we don't have to run it every time we run the tests.

Reply:
OK to comment them and leave a comment on why the might be useful.
Let's strive not to increase the test duration unless there's a tangible benefit.

If we ever want to support mingw binaries, we'd have better implementing a new ABI that more closely describes what mingw actually does, which seems to be radically different from what MSVC does (for 64-bits).

Reply:
Is this done?

Reply:
Yep, I just forgot to close the thread.

https://github.com/revng/revng-qa/pull/20/commits/4d89a0e59160b75fe2cbdb0b0bd7042b6b669ed6#diff-be7a1dce369aaf0e0d90ec71e4cca629675ac7af9b0c1d77f8e7d65509e850d2R167